### ISSUE TITLE
Learning weapon with split linked task

### ISSUE THREAD

**Issue by chrispikula** [jptrrs/HumanResources#95]
I really like the new system for weapon training, and how it is diverged away from techs.  Very nice!

However, I also found a strange, bug, that may be on my end.  I made 6 targets, all with a 1x do x times usage of a specific weapon, in this case, an EMP Grenade, and this task was not restricted to any particular pawn.  The bill was copied and connected to the other 5 targets.  (Better workbench management).  I only had 2 of these grenades in my colony, and I planned to make more.  Two pawns were forced to work on the tasks, and as soon as each one was done, there was an error thrown, the weapon was on the ground beside them, and they were stuck standing, not taking any more jobs.  Classic Object Reference not set to an instance of an object.  Not much detail, and at this point, I've sadly cleared the console (by reflex.  :-p)

As the first weapon was on the ground, another pawn tried to do the bill as well.  I paused, and forced the pawn who was standing to equip the weapon.  This caused the error, I believe.  The new pawn still walked to the target (a different target than the ones the first two pawns used), but this time they went to practice shooting, even though no practice shooting bill was in play.  This seems like a fallback for when a pawn can't do what they meant to do.  I like it!

The two pawns with grenades did learn the weapons, btw.

As an odd aside, they learned to use the weapon without actually using it, due to a conflict with Yayo's combat, and it's optional ammo system.  The grenades are made without ammo, and you can only arm them when you have them equipped.  Which you can't do before you train with them, at least not till a single pawn knows how, and you pass the loaded weapon around.

I'm holding off on submitting a log at the moment as I think my log is a bit too cluttered to be of any real use, what with >.5k mods.

**Comment by jptrrs** [comment]
I'm glad you liked the new system! :)
But for me, that reflex is bad! Even when cluttered, a snapshot of what was going on at the time is useful to figure things out. Worst case scenario its just inconclusive and I'd need to ask for further details. Now I know there's a possible error, but don't know what happens or when...  
BTW, they now go practicing on their own just for fun, even without the bill. If you feel that's interfering, you can actually turn it off at the mod options panel.

**Comment by chrispikula** [comment]
I know, it's kind of a bad reflex, but there is the compounding issue on my side that it's a silly amount of work to narrow it down.  :-/  30 minute load times are no joke.  Add to that the mistakes I've made in the past what with forgetting my config settings for all my mods, not getting rid of old config settings when paring down list, restarting worlds, and the hour or so to re-make config settings once lost turns it into a massive amount of wariness.  A single mistake can mean kaput, and those things tend to kill my enthusiasm to even play, never mind bug hunt.

I'll try to work on narrowing it down, but as it's relatively easy to work around and not a show stopper, it's hard to be as motivated.

I'm not asking for a fix, I should have pointed it out.  I guess it's more a 'I've noticed this, if anyone else see something similar we can start narrowing down possibilites' kind of thing.  I'll try to find a way to throw up a log with the error, but I tend to hit the max size on pastebin only an hour or two after starting a the game.
